src/braket/devices/device.py

```diff
@@ -128,21 +126,5 @@ def _validate_device_noise_model_support(self, noise_model: NoiseModel) -> None:
     def _apply_noise_model_to_circuit(
         self, task_specification: TaskSpecification
     ) -> TaskSpecification:
-        if isinstance(task_specification, Circuit):
-            for instruction in task_specification.instructions:
-                if isinstance(instruction.operator, Noise):
-                    warnings.warn(
-                        "The noise model of the device is applied to a circuit that already has"
-                        " noise instructions.",
-                        stacklevel=2,
-                    )
-                    break
-            task_specification = self._noise_model.apply(task_specification)
-        else:
-            warnings.warn(
-                "Noise model is only applicable to circuits. The type of the task specification is"
-                f" {task_specification.__class__.__name__}. The noise model of the device does not"
-                " apply.",
-                stacklevel=2,
-            )
-        return task_specification
+        """Deprecated in favor of NoiseModel.apply()"""
+        return self._noise_model.apply(task_specification)
```

Comment:
This is a private method, so why not just get rid of it altogether?

Reply:
+1